xdsl/xdsl_opt_main.py

```diff
@@ -164,6 +164,13 @@ def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
             help="Print operations with debug info annotation, such as location.",
         )
 
+        arg_parser.add_argument(
+            "--print-reduced-precision-fp",
+            default=False,
+            action="store_true",
+            help="Prints floating point literals with reduced precision.",
+        )
+
```

Comment:
I'm really not sure that this is a positive addition to the framework. I don't see why one would want to print reduced precision if the output isn't nicer. I'm happy to help with this PR, as it will will take some time, but I would feel much better if we didn't merge this flag.